### DISCUSSION TITLE
코드를 보고 arguments의 출력이 어떻게 될 지 결과를 설명해주세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#78] [실행컨텍스트]
- 코드를 보고 arguments의 출력이 어떻게 될 지 결과를 설명해주세요.
   (arguments 객체에 포함된 프로퍼티와 메서드의 종류에 대해서는 구체적으로 설명하지 않아도 됩니다.)
  ```js
  function a() {
      console.log(arguments);
  }
  
  const b = () => {
      console.log(arguments);
  }
  
  a();
  b();
  ```

**Comment by publdaze** [comment]
> - 코드를 보고 arguments의 출력이 어떻게 될 지 결과를 설명해주세요.
>   ```js
>   function a() {
>       console.log(arguments);
>   }
>   
>   const b = () => {
>       console.log(arguments);
>   }
>   
>   a();
>   b();
>   ```

- a가 호출되면, 함수 평가 과정의 거치며 렉시컬 환경을 구성하게 되는데, 함수선언문의 경우 매개변수, 지역변수와 함께 arguments 객체가 같이 저장이 되므로, arguments 객체의 프로퍼티와 메서드가 출력됨
- b는 화살표 함수로 정의되어 있는데, 화살표 함수의 경우 자신만의 렉시컬 환경을 생성하지 않기 때문에 arguments 객체를 가지는 환경레코드가 따로 존재하지 않고 이 arguments 객체를 찾기 위해 상위 스코프를 타고 올라가며 검색을 했을 때 상위 스코프에도 존재하지 않기 때문에 reference에러가 발생함

**Comment by pipisebastian** [comment]
> 코드를 보고 arguments의 출력이 어떻게 될 지 결과를 설명해주세요

- a함수는 함수 선언문으로, `arguments` 객체가 출력이됩니다.
- b함수는 화살표 함수로, 화살표 함수는 자신만의 `this`, `arguments` 바인딩을 가지지 않고, 상위 스코프의 렉시컬 환경을 사용합니다. 
- 상위 스코프인 전역 스코프에서도 `arguments`를 찾지만, 존재하지 않기 때문에 reference 에러가 발생합니다.
- +) 그러나 화살표 함수도 자신의 블록 스코프를 가지며, 내부에서 선언된 변수는 해당 블록 스코프에 속합니다.
